### PR TITLE
Add table for OTLP/HTTP response code and client retry recommendation

### LINKED ISSUE #2915: OTLP/HTTP: Retry or no on status code 401, 403?
**Is your feature request related to a problem? Please describe.**
401, 403 means auth failure, typically retry should not be attempted.

**Describe the solution you'd like**
add 401, 403 as `PermanentClientFailure` [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/otlphttpexporter/otlp.go#L189-L210)

### PR DESCRIPTION
**Pull request by newly12**
Fixes #2915

## Changes

Add table for OTLP/HTTP response code and client retry recommendation.

Related issues #2915

Related [OTEP(s)](https://github.com/open-telemetry/oteps) #

### CODE REVIEW

File: specification/protocol/otlp.md [hunk 1]
```diff
@@ -547,6 +547,27 @@ defined in `Status` message schema.
 The server MAY include `Status.details` field with additional details. Read
 below about what this field can contain in each specific failure case.
 
+The server is recommended to use HTTP response status codes to indicate
```
[the remark this answers is not in the document]
Comment (newly12) [reply specification/protocol/otlp.md]: Thanks for review.

**Review by tigrannajaryan — APPROVED** [review]

**Comment by jmacd** [comment]
Please add a changelog entry and a spec-compiance-matrix entry. Thanks.

**Review by carlosalberto — APPROVED** [review]

**Review by jack-berg — APPROVED** [review]
Please add a changelog entry. Thanks!

**Comment by newly12** [comment]
Not sure how spec-compiance-matrix should be updated, I suppose it's this row `Honors retryable responses with backoff` but I don't have much knowledge for all SDKs..

File: specification/protocol/otlp.md [hunk 2]
```diff
@@ -547,6 +547,27 @@ defined in `Status` message schema.
...
+|403 Forbidden|No|
+|404 Not Found|No|
+|405 Method Not Allowed|No|
+|413 Payload Too Large|No|
+|414 URI Too Long|No|
+|429 Too Many Requests|Yes|
+|431 Request Header Fields Too Large|No|
+|502 Bad Gateway|Yes|
+|503 Service Unavailable|Yes|
+|504 Gateway Timeout|Yes|
```
Comment (MrAlias) [inline specification/protocol/otlp.md:569]: This is not a complete list of status codes in the 400-599 range. By omitting status codes in the range here it adds ambiguity to the specification.

Instead, why not list the status codes in the range that can be retried and for the remaining give a general statement that they are not retryable?

Comment (newly12) [reply specification/protocol/otlp.md]: good point. updated.

Comment (carlosalberto) [reply specification/protocol/otlp.md]: @MrAlias Does this look good to you know?

File: specification/protocol/otlp.md [hunk 3]
```diff
@@ -547,6 +547,19 @@ defined in `Status` message schema.
 The server MAY include `Status.details` field with additional details. Read
 below about what this field can contain in each specific failure case.
 
+The server SHOULD use HTTP response status codes to indicate
+retryable and not-retryable errors for a particular erroneous situation. The
+client SHOULD honour HTTP response status codes as retryable or not-retryable,
+status codes listed in following table are retryable and the other `4xx` or
+`5xx` status codes are not retryable.
```
Comment (MrAlias) [inline specification/protocol/otlp.md:554]: Let's make this normative.

```suggestion
client SHOULD honour HTTP response status codes as retryable or not-retryable.
The requests that receive a response status code listed in following table SHOULD be retried.
All other `4xx` or `5xx` response status codes MUST NOT be retried.
```

File: specification/protocol/otlp.md [hunk 4]
```diff
@@ -547,6 +547,19 @@ defined in `Status` message schema.
...
+
+|HTTP response status code|Retryable?|
```
Comment (MrAlias) [inline specification/protocol/otlp.md:556]: Not sure the `Retryable` column is needed anymore.

Comment (newly12) [reply specification/protocol/otlp.md]: Given above already states there are retryable codes, I think it's not needed. Had removed it.

**Review by MrAlias — APPROVED** [review]

**Pull request merged**